src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -468,19 +468,18 @@ if this_event.type == pbs.PERIODIC:
             # nodes which are still booting, write them to
             # node file for status check in next iteration.
             if sleep_node_list or poweron_vnlist:
+                sleeping_nodes = poweron_vnlist
```

Comment:
Since you are doing an 'or' sleep_node_list or poweron_vlist can be None.  The for or the append() will fail if this is the case.  Are you sure you don't want an 'and'?

Reply:
We need or as both may not be true all the time. What you said about append() is right so I added an extra check.